pkg/kubelet/prober/results/results_manager.go

```diff
@@ -87,3 +115,26 @@ func (m *manager) Remove(id kubecontainer.ContainerID) {
 	defer m.Unlock()
 	delete(m.cache, id)
 }
+
+func (m *manager) Updates() <-chan Update {
+	updates := m.updates.Load().(chan Update)
+	if updates == nil {
+		m.Lock()
+		defer m.Unlock()
+		// Verify updatse did not change while we were aquiring the lock
+		updates = m.updates.Load().(chan Update)
+		if updates == nil {
+			updates = make(chan Update)
```

Comment:
sgtm. I think it'd be more than enough under regular circumstances. If it does become a bottleneck, that means something wrong in the sync loop itself and we probably need to fix that.